network/src/protocols/wire/handshake/v1/mod.rs

```diff
@@ -63,6 +63,7 @@ pub struct SupportedProtocols(bitvec::BitVec);
 pub struct HandshakeMsg {
     pub supported_protocols: BTreeMap<MessagingProtocolVersion, SupportedProtocols>,
     pub network_id: NetworkId,
+    pub chain_id: ChainId,
```

Comment:
per messaging earlier, think about the ordering here, cause in terms of evaluation, it would seem we first check universe (chain id), before we check galaxy (network id)... horrible analogy :P

Reply:
This was because of TOML for the ordering.  Once your PR goes through, we can reorder everything :)